src/main/java/me/gaminglounge/guiapi/Pagenation.java

```diff
@@ -108,22 +160,45 @@ public void fillPage(int pageNum) {
         try {
             for (int y = 0; y < 4; y++) {
                 for (int x = 0; x < 7; x++) {
-                    inv.setItem((y + 1) * 9 + x + 1, items.get(y*7 + x +pageNum*numItemsOnPage));
+                    inv.setItem((y + 1) * 9 + x + 1, items.get(y * 7 + x + pageNum * numItemsOnPage));
                 }
             }
-        } catch(IndexOutOfBoundsException | NullPointerException e) {}
+        } catch (IndexOutOfBoundsException | NullPointerException e) {
+        }
 
-        for (int a = inv.firstEmpty(); a<44; a++) {
-            if (a == -1) break;
+        for (int a = inv.firstEmpty(); a < 44; a++) {
+            if (a == -1)
+                break;
             if (a == 18 || a == 27 || a == 36 ||
-            a == 17 || a == 26 || a == 35) continue;
-            inv.setItem(a, placeholderItem);
+                    a == 17 || a == 26 || a == 35)
+                continue;
+            inv.setItem(a,
+                    new ItemBuilder(Material.GRAY_STAINED_GLASS_PANE)
+                            .setName(mm.deserialize(""))
+                            .addBothClickEvent("ItemBuilder:cancel")
+                            .build());
         }
 
     }
 
-    public void reFillPage(List<ItemStack> newItems) {
-        items = newItems;
+    /*
+     * Refreshed the page in case of item change
+     */
+    public void refeshPage() {
         fillPage(currentpage);
     }
+
+    /*
+     * adds an item to the list of items, refesh with: refeshPage
+     */
+    public void addItem(ItemStack item) {
```

Comment:
Should this be a javadoc comment?
```suggestion
    /**
     * adds an item to the list of items, refesh with: refeshPage
     */
    public void addItem(ItemStack item) {
```